libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -169,12 +166,27 @@ public FIXConnection(ReadableByteChannel inboundChannel, GatheringByteChannel ou
 
         this.adminMessage = new FIXMessage(ADMIN_MESSAGE_MAX_FIELD_COUNT, config.getFieldCapacity());
 
-        this.currentTimeMillis = 0;
-
         this.currentTime = new FIXTimestamp();
 
         this.currentTimestamp = new FIXValue(CURRENT_TIMESTAMP_FIELD_CAPACITY);
+        
+        initializeTime(System.currentTimeMillis());
```

Comment:
This is a pet peeve of mine - using system time directly in a library class can lead to non-reproducible tests.

In this case there is an easy workaround - call `initializeTime()` before you send the first message, but then the design allows you not to do it too.

One possible way is to add a constructor parameter. 

Another is to set the times to -1 and throw informative exception if time is not initialized properly when sending a message. This will make it slightly more cumbersome to use, but will achieve the following: 

1. time will be initialized with user-controlled value, making it more test friendly
2. would remind people who don't read docs that handling time is their responsibility (I admit I missed that the first time around)

Reply:
I agree, adding a constructor is a good and test friendly way to get rid of the public method.
As we are talking about a new major version, it is also legit to change some constructor parameters.

Regarding the informative exception I don't like checking proper initialization in a core function for performance reasons, even if we are talking about micro-optimizations. It would also require to keep the public `initializeTime()` method.

Reply:
The check can be an assertion, so those concerned about the last drop of performance can disable it (though they should be using FPGA by now). Pretty much all build tools run tests with assertions enabled, so that should be good enough.

Reply:
I implemented the new constructor [here](https://github.com/paritytrading/philadelphia/pull/302/commits/92f31939b3da519b57026813d15f0bf9f42ac043).
As written in the discussion about the testcase, the testcase is omited, as I think, one will think twice, before removing a parameter in the constructor of a core class. To reach the main goal of the test, which is to prevent the reintroduction of the inproper initialization of the time variables, the new parameter should do the same job.